static/javascript/map_draw.js

```diff
@@ -612,17 +628,304 @@ const createDistanceTooltip = (lat, lng, distanceKm) => {
     .setLatLng(new L.LatLng(lat, lng));
 };
 
-// TODO: clean up this function
-// TODO: record better route overall, not just between current and previous point
-let routeData = [
-  /*
+/** For a location find nearby segments using the spatial index. */
```

Comment:
This whole section of new lines is refactoring out from the previous `snapToNetwork` function.